test/unit/org/apache/cassandra/index/sai/cql/types/IndexingTypeSupport.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.index.sai.cql.types;
+
+import java.util.Arrays;
+import java.util.Collection;
+
+import org.junit.Before;
+
+import org.apache.cassandra.index.sai.SAITester;
+
+public abstract class IndexingTypeSupport extends SAITester
+{
+    public static final int NUMBER_OF_VALUES = 64;
+
+    protected final DataSet<?> dataset;
+
+    private final boolean widePartitions;
+    private final Scenario scenario;
```

Comment:
We can make these three public and use the `@Parameter` annotation:
```
@Parameterized.Parameter
public DataSet<?> dataset;

@Parameterized.Parameter(1)
public boolean widePartitions;

@Parameterized.Parameter(2)
public Scenario scenario;
```
That way we could get rid of the constructor that takes those tree parameters, and use the default constructor instead. Then, the 101 subclasses of `IndexingTypeSupport` wouldn't need to define a constructor just calling to super's constructor. That would save us a few lines of repetitive code on each subclass. Combined with the other suggestions about moving the test method and the runner annotation to `IndexingTypeSupport`, we can save us more than a thousand lines, and reduce the code in the subclasses to a minimum.

Reply:
Agreed, this does make the tests a lot simpler in definition